### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by stephenkati**
I used a design by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

### Interactions
**Splash screen**

- A simple page with the name of your app, and links to the sign up and log in pages.

**Sign up and log in pages**

- The user should be able to register in the app with full name, email and password (all mandatory).

- The user can log into the app using email and password.

- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (categories page)**

- When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

- The button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.

- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"new category" page**

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).

- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"new transaction" page**

- The user fills out a form to create a new transaction with:

1. name (mandatory)
2. amount (mandatory)
3. categories (mandatory at least one)

- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing**

- Created unit and integration tests for all the most important components of your RoR application.

**Important technical concepts**

- Used `Postgres` as your database.
- Used `devise` for authentication.
- Validated all user input to make sure that anyone with bad intentions cannot compromise your app.
- Used a view template engine (.erb).
- The project is deployed and accessible online via this link (  )

**My database schema should reflect the following structure:**
https://github.com/microverseinc/curriculum-rails/blob/main/capstone/images/erd_diagram.png

**Project documentation**
Here is a [video](https://www.loom.com/share/7b3a446fdd954b49b56ea513efdfc865) presenting:

- a description of the project.
- a demo of the project features.
- highlight of interesting piece of code / something I built that I am very proud of.

### CODE REVIEW

File: app/views/users/splash.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="w-full h-full" >
+
+  <div class="h-2/3 flex items-center justify-center">
+    <h2 class="font-bold text-primary text-2xl"> Smart Wallet </h2>
+  </div>
+
+  <div class="flex flex-col items-center justify-center gap-2">
+    <%= button_to "LOG IN", new_user_session_path, class: "bg-primary text-white px-7 py-2" %>
+    <%= button_to "SIGN UP", new_user_registration_path, class: "hover:bg-neutral-focus px-7 py-2" %>
+  </div>
```
Comment (andersonlebon) [inline app/views/users/splash.html.erb:10]: - [ ] As the requirements say, you design should look exactly the same as the one from the template. Kindly make sure you apply the same here on this welcome page, use the same color, fontfamily, font-size, layout,...

Your design | the template 
---|----
image | image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <%= f.password_field :password_confirmation, autocomplete: "new-password", required: true, class: "rounded-full border-none w-3/4 bg-neutral text-white" %>
+      </div>
+
+      <div class="font-bold bg-secondary hover:bg-secondary-focus px-4 py-1">
+        <%= f.submit "Sign up" %>
+      </div>
+    <% end %>
+
+    <%= render "devise/shared/links" %>
+  </div>
```
Comment (andersonlebon) [inline app/views/devise/registrations/new.html.erb:40]: - [ ] As the requirements say, you design should look exactly the same as the one from the template. Kindly make sure you apply the same here on the login and signup pages, use the same color, fontfamily, font-size, layout,...

Your design | the template 
---|----
image | image
image | image

File: app/views/categories/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            </div>
+          </div>
+          <span class="text-accent"> <%= "$#{category.total_amount}" %> </span>
+        <% end %>
+
+      </li>
+
+    <% end %>
+
+  </ul>
```
Comment (andersonlebon) [inline app/views/categories/index.html.erb:42]: - [ ] As the requirements say, you design should look exactly the same as the one from the template. Kindly make sure you apply the same here on the index and new category  pages, use the same color, fontfamily, font-size, layout,...

Your design | the template 
---|----
image | image

image | image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @stephenkati 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

almost-there

## Highlights 🍾 

- Nice work on using tailwinds 👏  ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/users/splash.html.erb (hunk 1 above)
Comment (stephenkati) [reply app/views/users/splash.html.erb:10]: you are correct, my design does not look like the behance design. Thanks for the correction

File: app/views/categories/index.html.erb (hunk 3 above)
Comment (stephenkati) [reply app/views/categories/index.html.erb:42]: I see my mistake, thanks for the correction

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @stephenkati,

✔️ Nice UI designs

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**